enterprise/server/util/cacheproxy/cacheproxy.go

```diff
@@ -123,6 +125,13 @@ func (c *CacheProxy) getClient(ctx context.Context, peer string) (dcpb.Distribut
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	if client, ok := c.clients[peer]; ok {
```

Comment:
I think you need to add logic to force a reconnect if the connection is idle.
While the keepalive changes will help normal connections from idling out, a broken connection will still put the connection nito IDLE state and it will stay there because we won't send any further RPCs.

Reply:
Yup, done.